### PR TITLE
Internal: Documentation of `document/history` component. [ED-6496]

### PR DESCRIPTION
**Pull request by iNewLegend**

### CODE REVIEW

File: docs/assets/dev/js/editor/document/history/readme.md [hunk 1]
```diff
@@ -0,0 +1,145 @@
+## Component -- `$e.components.get('document/history')`
+
+*  **Name**: History.
+*  **Description**: Provide a way to manage history, create/delete history items, undo and redo history items, add transactions to be saved as history item.
```
Comment (ItzhakAvm) [inline docs/assets/dev/js/editor/document/history/readme.md:4]:
```suggestion
*  **Description**: Provide a way to manage editor history, which means - to create and delete history items (milestones), time-travel between them (undo and redo), and starting transactions to be saved as history items.
```

File: docs/assets/dev/js/editor/document/history/readme.md [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+
+## _Command_ -- `$e.run('document/histroy/redo')`
+*  **Name**: Redo.
+*  **Description**: Redo history step.
+*  **Returns**: `{void}`
+*  **Arguments**: None.
+
+## _Command_ -- `$e.internal('document/histroy/add-transaction')`
+*  **Name**: Add-Transaction.
+*  **Description**: Add item to transactions.
```
Comment (ItzhakAvm) [inline docs/assets/dev/js/editor/document/history/readme.md:72]: Maybe explain a little bit about the differences between items, logs and transactions?

**Review by YitzhakAvm — CHANGES_REQUESTED** [review]

File: docs/assets/dev/js/editor/document/history/readme.md (hunk 2 above)
Comment (iNewLegend) [reply docs/assets/dev/js/editor/document/history/readme.md:79]: Handled.

**Review by YitzhakAvm — APPROVED** [review]

File: docs/assets/dev/js/editor/document/history/readme.md [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+|-----------------------------------------------------------------------------|-----------------------------------------------------|-----------------------------------------
+| [Add-Transaction](#command----einternaldocumenthistroyadd-transaction)      | `$e.internal('document/history/add-transaction')`   | Add transaction item.
+| [Clear-Transaction](#command----einternaldocumenthistroyclear-transaction)  | `$e.internal('document/history/clear-transaction')` | Clear transaction.
+| [Delete-Log](#command----einternaldocumenthistroydelete-log)                | `$e.internal('document/history/delete-log')`        | Delete log.
+| [End-Log](#command----einternaldocumenthistroyend-log)                      | `$e.internal('document/history/end-log')`           | End log.
+| [End-Transaction](#command----einternaldocumenthistroyend-transaction)      | `$e.internal('document/history/end-transaction')`   | End Transaction.
+| [Log-Sub-Item](#command----einternaldocumenthistroylog-sub-item)            | `$e.internal('document/history/log-sub-item')`      | Lob sub item.
```
Comment (StyleShit) [inline docs/assets/dev/js/editor/document/history/readme.md:47]:
```suggestion
| [Log-Sub-Item](#command----einternaldocumenthistroylog-sub-item)            | `$e.internal('document/history/log-sub-item')`      | Log sub item.
```

File: docs/assets/dev/js/editor/document/history/readme.md [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+
+
+## Component `document/history/` -- Internal Commands
+| Command                                                                     | Access                                              | Description
+|-----------------------------------------------------------------------------|-----------------------------------------------------|-----------------------------------------
+| [Add-Transaction](#command----einternaldocumenthistroyadd-transaction)      | `$e.internal('document/history/add-transaction')`   | Add transaction item.
+| [Clear-Transaction](#command----einternaldocumenthistroyclear-transaction)  | `$e.internal('document/history/clear-transaction')` | Clear transaction.
+| [Delete-Log](#command----einternaldocumenthistroydelete-log)                | `$e.internal('document/history/delete-log')`        | Delete log.
+| [End-Log](#command----einternaldocumenthistroyend-log)                      | `$e.internal('document/history/end-log')`           | End log.
+| [End-Transaction](#command----einternaldocumenthistroyend-transaction)      | `$e.internal('document/history/end-transaction')`   | End Transaction.
```
Comment (StyleShit) [inline docs/assets/dev/js/editor/document/history/readme.md:46]:
```suggestion
| [End-Transaction](#command----einternaldocumenthistroyend-transaction)      | `$e.internal('document/history/end-transaction')`   | End transaction.
```

File: docs/assets/dev/js/editor/document/history/readme.md [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+*  **Name**: Add-Transaction.
+*  **Description**: Add item to transactions.
+*  **Returns**: `{void}`
+*  **Arguments**:
+
+   | Property     | Type                  | Requirement   | Description |
+   |---           |---                    |---            |---|
+   | _container_  | `{Container}`         | **require**   | Container log.
+   | _containers_ | `{Container[]}`       | **require**   | Containers log.
+   | _type_       | `{String}`            | **require**   | Type
```
Comment (StyleShit) [inline docs/assets/dev/js/editor/document/history/readme.md:87]:
```suggestion
   | _container_  | `{Container}`         | **required**   | Container log.
   | _containers_ | `{Container[]}`       | **required**   | Containers log.
   | _type_       | `{String}`            | **required**   | Type
```

File: docs/assets/dev/js/editor/document/history/readme.md [hunk 6]
```diff
@@ -0,0 +1,152 @@
...
+## _Command_ -- `$e.internal('document/histroy/end-transaction')`
+*  **Name**: End-Transaction.
+*  **Description**: End transaction, will log the first and the last transaction, as new logged history item.
+   title, subTitle will be taken from the first transaction item.
+*  **Returns**: `{void}`
+*  **Arguments**: None.
+
+## _Command_ -- `$e.internal('document/histroy/log-sub-item')`
+*  **Name**: Log-Sub-Item.
+*  **Description**: Log sub item, Each history item can have sub items ( non visual at history panel ).
```
Comment (StyleShit) [inline docs/assets/dev/js/editor/document/history/readme.md:123]:
```suggestion
*  **Description**: Log sub-item. Each history item can have sub-items (not visualized in the history panel).
```

File: docs/assets/dev/js/editor/document/history/readme.md [hunk 7]
```diff
@@ -0,0 +1,152 @@
...
+*  **Description**: Start log item.
+*  **Returns**: `{Number}` *log id*.
+*  **Arguments**:
+
+   | Property     | Type                  | Requirement   | Description |
+   |---           |---                    |---            |---|
+   | _container_  | `{Container}`         | **require**   | Container log.
+   | _containers_ | `{Container[]}`       | **require**   | Containers log.
+   | _type_       | `{String}`            | **require**   | Type
+   | _title_      | `{String}`            | **require**   | Title.
```
Comment (StyleShit) [inline docs/assets/dev/js/editor/document/history/readme.md:148]:
```suggestion
   | _container_  | `{Container}`         | **required**   | Container log.
   | _containers_ | `{Container[]}`       | **required**   | Containers log.
   | _type_       | `{String}`            | **required**   | Type
   | _title_      | `{String}`            | **required**   | Title.
```

**Review by StyleShit — CHANGES_REQUESTED** [review]

**Review by StyleShit — APPROVED** [review]

**Pull request merged**